File: app/src/main/java/com/mikhailovskii/trakttv/ui/profile/ProfileContract.java
package com.mikhailovskii.trakttv.ui.profile;

import androidx.annotation.NonNull;

import com.mikhailovskii.trakttv.data.entity.User;
import com.mikhailovskii.trakttv.ui.base.MvpPresenter;
import com.mikhailovskii.trakttv.ui.base.MvpView;

public interface ProfileContract {

    interface ProfileView extends MvpView {

        void onLogOutSuccess();

        void onUserDataLoaded(@NonNull User user);

    }

    interface ProfilePresenter extends MvpPresenter<ProfileView> {

        void getUser();

        void logOut();

    }

}

File: app/src/main/java/com/mikhailovskii/trakttv/ui/profile/ProfileContract.kt
package com.mikhailovskii.trakttv.ui.profile

import com.mikhailovskii.trakttv.data.entity.User
import com.mikhailovskii.trakttv.ui.base.MvpPresenter
import com.mikhailovskii.trakttv.ui.base.MvpView

interface ProfileContract {

    interface ProfileView : MvpView {

        fun onLogOutSuccess()

        fun onUserDataLoaded(user: User)

    }

    interface ProfilePresenter : MvpPresenter<ProfileView> {

        fun getUser()

        fun logOut()

    }

}
